content/articles/transaction-management-in-database/index.md.md

```diff
@@ -0,0 +1,220 @@
+5. Terminated state- This state happens once the transaction leaving the system cannot be restarted once again.
+
+### Transaction properties 
+There are four main properties of a transaction represented in the acronym ACID referring to; Atomicity, Consistency, Isolation, and Durability.
+
+1. Atomicity- A transaction cannot be subdivided and can only be executed as a whole and is treated as an atomic unit. It is either all the operations are carried out or none are performed.
+2. Consistency- After any transaction is carried out in a database it should remain consistent. No transaction should affect the data residing in the database adversely.
+3. Isolation- When several transactions need to be conducted in a database at the same time, each transaction is treated as if it were a single transaction. As a result, the completion of a single transaction should have no bearing on the completion of additional transactions.
+4. Durability- From durable, all changes made must be permanent such that once the transaction is committed the effects of the transaction cannot be reversed. In case of system failure or unexpected shutdown and changes made by a complete transaction are not written to the disk, during restart the changes should be remembered and restored.
+
+### Creating a database and a table
+We need to create a database with a table in which to execute our transaction commands.
+Here is the program to create a table;
+```SQL
+CREATE DATABASE MEGA COLLEGE;
+CREATE TABLE STUDENTS(Id integer PRIMARY KEY, Name text, Regno varchar, Course text, Fees integer);
+
+
+INSERT INTO STUDENTS VALUES(1,'John', 'V20/680', 'Education', 40000);
+INSERT INTO STUDENTS VALUES(2,'Ahmed', 'G4/270', 'Computer Science',50000);
+INSERT INTO STUDENTS VALUES(3,'Ian', 'A43/476', 'Engineering', 62500);
+INSERT INTO STUDENTS VALUES(4,'Jane', 'A19/520', 'IT', 41200);
+INSERT INTO STUDENTS VALUES(5,'Robert','P62/491', 'Nursing',43700);
+INSERT INTO STUDENTS VALUES(6,'Hellen', 'V20/470', 'Education', 22000);
+INSERT INTO STUDENTS VALUES(7,'Joan','P62/801', 'Nursing',18000);
+COMMIT;
+
+
+SELECT * FROM STUDENTS;
+```
+Below is the output of the program.
+
+|ID | NAME   | REG NO  | COURSE     | FEES  |
+|-- | :-----:| :------:| :---------:| -----:|
+|1  | JOHN   |V20/630|Education|40000|
+|2  | AHMED  |G4/270|Computer science|50000|
+|3  | IAN    |A43/476|Engineering|62500|
+|4  | JANE|A19/520|IT|41200|
+|5  | ROBERT |P62/491|Nursing|43700|
+|6  | HELLEN |V20/470|Education|22000|
+|7  | JOAN   |P62/801|Nursing|18000|
+
+
+### Transactional control commands
+The Data Manipulation Language (DML) instructions are utilized with the transactional commands such as INSERT, UPDATE, and DELETE. These commands cannot be used with the Data Definition Language because they are automatically committed to the database. Below are the transaction commands; 
```

Comment:
```suggestion
The Data Manipulation Language (DML) instructions are utilized with transactional commands such as INSERT, UPDATE, and DELETE. These commands cannot be used with the Data Definition Language because they are automatically committed to the database. Below are the transaction commands; 
```